Sources/ArgumentParser/Completions/BashCompletionsGenerator.swift

```diff
@@ -9,257 +9,397 @@
+                  unset "unparsed_words[${word_index}]"
+                  # Do not process this word as a flag or an option
+                  continue
+              fi
+
+              local word="${unparsed_words["${word_index}"]}"
+              if ! "${was_flag_option_terminator_seen}"; then
+                  case "${word}" in
+                  --)
+                      unset "unparsed_words[${word_index}]"
+                      # by itself -- is a flag/option terminator, but if it is the last word, it is the start of a completion
+                      if [[ "${word_index}" -ne "${unparsed_word_indices[${#unparsed_word_indices[@]} - 1]}" ]]; then
+                          was_flag_option_terminator_seen=true
+                      fi
+                      continue
+                      ;;
+                  -*)
+                      # ${word} is a flag or an option
+                      # If ${word} is an option, mark that the next word to be parsed is an option value
+                      # TODO: handle joined-value options (-o=file.ext), stacked flags (-aBc), legacy long (-long), combos
+                      # TODO: if multi-valued options can exist, support them
+                      local option
+                      for option in "${options[@]}"; do
+                          [[ "${word}" = "${option}" ]] && is_parsing_option_value=true && break
+                      done
+
+                      # Remove ${word} from ${flags} or ${options} so it isn't offered again
+                      # TODO: handle repeatable flags & options
+                      # TODO: remove equivalent options (-h/--help) & exclusive options (--yes/--no)
+                      local not_found=true
+                      local -i index
+                      for index in "${!flags[@]}"; do
+                          if [[ "${flags[${index}]}" = "${word}" ]]; then
+                              unset "flags[${index}]"
+                              flags=("${flags[@]}")
+                              not_found=false
+                              break
+                          fi
+                      done
+                      if "${not_found}"; then
+                          for index in "${!options[@]}"; do
+                              if [[ "${options[${index}]}" = "${word}" ]]; then
+                                  unset "options[${index}]"
+                                  options=("${options[@]}")
+                                  break
+                              fi
+                          done
+                      fi
+                      unset "unparsed_words[${word_index}]"
+                      continue
+                      ;;
+                  esac
+              fi
+
+              # ${word} is neither a flag, nor an option, nor an option value
+              # TODO: can SAP be configured to require options before positionals?
```

Comment:
This isn't a feature currently – options/flags can come anywhere in the list of arguments.